File: src/main/java/org/blacksun/graph/algorithms/PathFindingAlgorithm.java
package org.blacksun.graph.algorithms;

import org.blacksun.graph.node.GraphNode;
import org.jetbrains.annotations.NotNull;

import java.util.HashMap;

public interface PathFindingAlgorithm {
    Integer INFINITY = Integer.MAX_VALUE;

    boolean isAccessible(@NotNull GraphNode from, @NotNull GraphNode to);
    int getMinDistance(@NotNull GraphNode from, @NotNull GraphNode to);
    HashMap<GraphNode, Integer> getDistances(@NotNull GraphNode node);
    GraphPath getPath(@NotNull GraphNode from, @NotNull GraphNode to);
}

File: src/main/java/org/blacksun/graph/algorithms/PathFindingAlgorithm.kt
package org.blacksun.graph.algorithms

import org.blacksun.graph.node.GraphNode

import java.util.HashMap

interface PathFindingAlgorithm {
    fun isAccessible(from: GraphNode, to: GraphNode): Boolean
    fun getMinDistance(from: GraphNode, to: GraphNode): Int
    fun getDistances(node: GraphNode): HashMap<GraphNode, Int>
    fun getPath(from: GraphNode, to: GraphNode): GraphPath

    companion object {
        const val INFINITY = Integer.MAX_VALUE
    }
}
